src/Inciter/AMR/tet_store.h

```diff
@@ -0,0 +1,733 @@
+                            // Note: Below here isn't used right now?
+                            case Refinement_Case::two_to_eight:
+                                val = 2.8;
+                                break;
+                            case Refinement_Case::four_to_eight:
+                                val = 4.8;
+                                break;
+                            case Refinement_Case::none:
+                                val = 0.0;
+                                break;
+                        }
+                        cell_type_list.push_back(val);
+                    }
+                }
+
+                trace_out << "Made cell type list of len " << cell_type_list.size() << std::endl;
+                return cell_type_list;
+            }
+
+            /**
+             * @brief Function to extract only the active elements from tetinpeol
+             *
+             * @return List of only active elements
+             */
+            // TODO: need equiv for m_x/m_y/m_z? Otherwise there will be
+            // useless nodes in m_x etc? (Although that's functionally fine)
+            std::vector< std::size_t >& get_active_inpoel()
+            {
+                active_tetinpoel.clear();
+                active_nodes.clear();
+
+                for (const auto& kv : tets)
+                {
+
+                    size_t element_id = kv.first;
+                    auto t = kv.second;
+
+                    if (is_active( element_id )) {
+                        active_tetinpoel.push_back( t[0] );
+                        active_tetinpoel.push_back( t[1] );
+                        active_tetinpoel.push_back( t[2] );
+                        active_tetinpoel.push_back( t[3] );
+
+                        active_nodes.insert( t[0] );
+                        active_nodes.insert( t[1] );
+                        active_nodes.insert( t[2] );
+                        active_nodes.insert( t[3] );
+                    }
+                }
+
+                return active_tetinpoel;
+            }
+
+            // TODO: Remove this as it shouldn't make a difference..but it's good to check
+            std::vector< real_t > get_active_coordinate_array(std::vector<real_t> in)
+            {
+                std::vector< real_t > active;
+
+                for (size_t i = 0; i < in.size(); i++)
+                {
+                    // If an active tet has an edge which contains this i value
+                    if ( active_nodes.find(i) != active_nodes.end() )
+                    {
+                        active.push_back( in[i] );
+                    }
+                }
+
+                trace_out << "in size " << in.size() << " active size " << active.size() << std::endl;
+                return active;
+            }
+
+            // TODO: These mark methods can probably be a single one to which a
+                // Refinement_Case is passed, depending on how extra edges are marked
+            /**
```

Comment:
Yeah, it seems like it would be good to do that to simplify and shorten the code follows.